services/supervisor/examples/configuration_change.rs

```diff
@@ -0,0 +1,169 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use exonum::{
+    blockchain::InstanceCollection,
+    helpers::Height,
+    messages::Verified,
+    runtime::{
+        rust::{CallContext, Service, Transaction},
+        AnyTx, BlockchainData, ExecutionError, InstanceId, SnapshotExt, SUPERVISOR_INSTANCE_ID,
+    },
+};
+use exonum_crypto::Hash;
+use exonum_derive::*;
+use exonum_merkledb::{
+    access::{Access, AccessExt},
+    Entry, ObjectHash, Snapshot,
+};
+use exonum_supervisor::{ConfigPropose, ConfigVote, Configure, DecentralizedSupervisor};
+use exonum_testkit::{TestKit, TestKitBuilder};
+
+const SERVICE_ID: InstanceId = 256;
+const SERVICE_NAME: &str = "config";
+
+// Simple service to provide an example of how to implement the service configuration change.
+#[derive(Clone, Default, Debug, ServiceFactory, ServiceDispatcher)]
+#[service_dispatcher(implements("ConfigChangeInterface", "Configure<Params = String>"))]
+#[service_factory(artifact_name = "config", artifact_version = "1.0.0")]
+pub struct ConfigChangeService;
+
+#[exonum_interface]
+pub trait ConfigChangeInterface {}
+
+#[derive(Debug, FromAccess)]
+pub struct Schema<T: Access> {
+    params: Entry<T::Base, String>,
+}
+
+impl Service for ConfigChangeService {
+    fn state_hash(&self, _data: BlockchainData<'_, &dyn Snapshot>) -> Vec<Hash> {
+        vec![]
+    }
+}
+
+impl ConfigChangeInterface for ConfigChangeService {}
+
+// To allow service change its configuration we need to implement `Configure` trait.
+impl Configure for ConfigChangeService {
+    type Params = String;
+
+    fn verify_config(
```

Comment:
Maybe add something like `println!("Verify config called with params {}", params)` here and in `apply_config` to make this example more example-ish?